### PR TITLE
Add PHPUnit as a dev dependency

### PR DESCRIPTION
**Pull request by theofidry**
PHPUnit was installed only as a transient dependency but as it is used for tests, it should be a dev dependency instead.

### CODE REVIEW

**Comment by bamarni** [comment]
This was a way to have a simple functional test, on each build the current project is added as a plugin (https://github.com/bamarni/composer-bin-plugin/blob/master/.travis.yml#L24-L25) and there is a phpunit namespace in the project.

Your change is still conflicting with https://github.com/bamarni/composer-bin-plugin/blob/74c748ca9db5980f17e5e5947565609cf5bd2ae3/composer.json#L23. If you want to change this could you still keep a few lines in travis.yml for a simple functional test? Even though it's not on the main project maybe a quick install on a tmp folder then checking ./bin/phpunit --version or something similar.

**Comment by theofidry** [comment]
@bamarni here's the change with the functional test. The reason why I wouldn't promote using phpunit in vendor-bin is because PHPUnit *executes* your code, and technically if it can use a different dep (e.g. `symfony/yaml`) different from your project one which can lead to nasty bugs. That said here there is no risk of that happening so it's up to you if you want to do it the "correct" way or if you want to be a bit more pragmatic and save a bit of time on the project.

Another reason why I would do it that way is that your scripts are run for each consumer. As a result, if I install this plugin, this plugin vendor-bin deps will be installed as well.

File: .travis.yml [hunk 1]
```diff
@@ -20,15 +20,14 @@ matrix:
   allow_failures:
     - php: nightly
 
-before_install:
-  - composer global config repositories.bin path $PWD
-  - composer global require bamarni/composer-bin-plugin:dev-master
-
 install:
   - composer install
+  - composer global require "bamarni/composer-bin-plugin:dev-master"
```
Comment (theofidry) [inline .travis.yml]: makes sense, so I'm just missing the `composer global config repositories.bin path $PWD` right?

Comment (bamarni) [inline .travis.yml]: yep, just before this line

**Comment by theofidry** [comment]
@bamarni done

**Comment by theofidry** [comment]
Useless now

**Pull request closed without merging**